api/admin/controller/catalog_services.py

```diff
@@ -67,24 +73,65 @@ def process_post(self):
             service.name = name
 
         [protocol] = [p for p in self.protocols if p.get("name") == protocol]
+
         result = self._set_integration_settings_and_libraries(service, protocol)
         if isinstance(result, ProblemDetail):
             return result
 
+        external_integration_link = self._set_external_integration_link(service)
+        if isinstance(external_integration_link, ProblemDetail):
+            return external_integration_link
+
         library_error = self.check_libraries(service)
         if library_error:
             self._db.rollback()
             return library_error
 
-        storage_protocol_error = self.check_storage_protocol(service)
-        if storage_protocol_error:
-            self._db.rollback()
-            return storage_protocol_error
-
         if is_new:
             return Response(unicode(service.id), 201)
         else:
             return Response(unicode(service.id), 200)
+    
+    def _set_external_integration_link(self, service):
+        mirror_integration_id = flask.request.form.get('mirror_integration_id')
+        if mirror_integration_id == self.NO_MIRROR_INTEGRATION:
+            current_integration_link = get_one(
+                self._db,
+                ExternalIntegrationLink,
+                library_id=None,
+                external_integration_id=service.id,
+                purpose="MARC"
+            )
```

Comment:
This bit is repeated in _update_external_integration_link -- it should be refactored. Since this would leave _update_external_integration_link almost empty, you could maybe replace it with _get_external_integration_link and move the code that sets the new value into this method.